protocols/encodings/encodings.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encodings
+
+import "fmt"
+
+// Encoding is the format that a protocol is serialized to.
+type Encoding string
+
+const (
+	Protobuf Encoding = "protobuf"
+	JSON     Encoding = "json"
+	Thrift   Encoding = "thrift"
+)
+
+func (e Encoding) String() string {
+	return string(e)
+}
+
+type ErrInvalidEncoding struct {
+	Encoding Encoding
+}
+
+func (e *ErrInvalidEncoding) Error() string {
+	return fmt.Sprintf("unsupported encoding %q", e.Encoding)
+}
```

Comment:
Not sure how this is useful. Is this for "unavailable" encoding? Please comment.

Reply:
Yes, renamed ErrUnavailableEncoding.